apps/passport-server/src/services/telegramService.ts

```diff
@@ -187,15 +154,22 @@ export class TelegramService {
     this.bot.on("chat_member", async (ctx) => {
       try {
         const newMember = ctx.update.chat_member.new_chat_member;
-        if (newMember.status === "member") {
+        if (newMember.status === "left" || newMember.status === "kicked") {
```

Comment:
Now, we only delete a verification when a user leaves or is banned. This means that we have an accurate list of everyone who is currently in the group. Previously, once people joined, we deleted their entry.